setup.py

```diff
@@ -0,0 +1,57 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+import os
+import sys
+try:
+    import skbuild
+except ImportError:
+    print("Trying to install required module: skbuild")
+    os.system(f"{sys.executable} -m pip install scikit-build")
+import skbuild
+import setuptools
+
+__version__ = "0.0.3"
```

Comment:
This should be in sync with the CUDA Quantum version. Would it make sense to have this be picked up from an environment variable, or does the version number need to be in the setup.py file? Either way, we'll ultimately update this to be automatically picked up during build rather than hardcoded here (not in this PR, but please change this to 0.4.0 for now).